pvlib/tests/test_irradiance.py

```diff
@@ -792,6 +792,14 @@ def test_aoi_and_aoi_projection(surface_tilt, surface_azimuth, solar_zenith,
     assert_allclose(aoi_projection, aoi_proj_expected, atol=1e-6)
 
 
+def test_aoi_projection_precision():
+    # GH 1185
+    zenith = 89.26778228223463
+    azimuth = 60.932028605997004
+    projection = irradiance.aoi_projection(zenith, azimuth, zenith, azimuth)
+    assert projection == 1
```

Comment:
I don't know enough to intelligently comment on this, but I noticed that we're clipping to the `int` values `-1` and `1`. Is `projection` here also an int? I believe that if tested with realistic array input we'd always see floats returned. Might be worth testing that the clip behavior remains as expected in that case. Perhaps the important thing is that projection is always <= 1 and that np.isclose(projection, 1) returns True.

Reply:
Good suggestions.  I also added a dtype check to make sure that clipping with `int` limits doesn't do something funky.